napari/components/viewer_model.py

```diff
@@ -714,8 +715,10 @@ def open_sample(
             raise KeyError(msg)
 
         if callable(data):
+            mthd = f'{type(self).__module__}.{type(self).__name__}.open_sample'
             for datum in data(**kwargs):
-                self._add_layer_from_data(*datum)
+                for i in self._add_layer_from_data(*datum):
+                    i._source = Source(path=sample, plugin=plugin, method=mthd)
```

Comment:
Is the plugin here the plugin that provided the sample data, or the reader plugin that got used? Looks like it is the sample data plugin, I worry though about it being ambiguous in this case, i could see arguments for the reader plugin

Reply:
If a sample data hookspec provides a function (not a string), there _is no_ reader plugin involved.  So I used the sample data plugin.  If it’s a string, then it goes to viewer.open... where the reader plugin will be used

Reply:
ahhh, right! thanks for the clarification there, this is the right choice then!!